lambda/utilities/common_functions.py

```diff
@@ -279,14 +280,14 @@ def get_id_token(event: dict) -> str:
     auth_header = None
 
     if "authorization" in event["headers"]:
-        auth_header = event["headers"]["authorization"].split(" ")
+        auth_header = event["headers"]["authorization"]
     elif "Authorization" in event["headers"]:
-        auth_header = event["headers"]["Authorization"].split(" ")
+        auth_header = event["headers"]["Authorization"]
     else:
         raise ValueError("Missing authorization token.")
 
-    token = auth_header[1]
-    return str(token)
+    # remove bearer token prefix if present
+    return re.sub(r"Bearer ", "", str(auth_header), flags=re.I).strip()
```

Comment:
I'd prefer the removeprefix string method since it's a bit easier to read than this and doesn't import a library, but not a blocking comment. this still does the job- it's a stretch, but this may be a problem if somehow a user has a token that also contains "Bearer " in it (largely unlikely, but possible)

```
>>> import re
>>> re.sub(r"Bearer ", "", "Bearer Bearer 1", flags=re.I)
'1'
```

vs

```
>>> "Bearer Bearer 1".removeprefix("Bearer ")
'Bearer 1'
```

something to consider, i'll accept either way though- i'm sure secrets manager wouldn't give us that string specifically, and spaces probably shouldn't make it into an auth header definition

Reply:
I'll push the change with `removeprefix` but since there is no capitalization guarantees so it is actually between the two below.

```
>>> import re
>>> re.sub(r"^Bearer ", "", id_token, flags=re.I)
``` 

and

```
>>> id_token.removeprefix("Bearer ").removeprefix("bearer ")
```